### PR TITLE
update to use go1.13.4

### PR DESCRIPTION
**Pull request by liggitt**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
* Updates to go1.13.4
  * Updates to [bazel rules_go 0.20.2](https://github.com/bazelbuild/rules_go/releases/tag/0.20.2)
  * Updates to [gazelle 0.19.0](https://github.com/bazelbuild/bazel-gazelle/releases/tag/0.19.0)
  * Updates to [buildtools 0.29.0](https://github.com/bazelbuild/buildtools/releases/tag/0.29.0)
* Updates golang.org/x/... dependencies to go1.13 levels
* Updates to github.com/go-openapi/jsonpointer v0.19.3 to pick up https://github.com/go-openapi/jsonpointer/pull/4, required for 1.13
* Updates k8s.io/repo-infra to pick up https://github.com/kubernetes/repo-infra/pull/150, required to make BUILD file generation consistent
* Updates minimum go version to 1.13.4
  * Required for consistent vendor generation, and for bazel to build golang.org/x/crypto
* Continues to build in GOPATH mode (not module mode). This is to minimize build changes so we can pick go1.13 support to release-1.16. Updates to start using modules will be done in follow-ups.

Before merge:
- [x] Wait for fix for https://github.com/golang/go/issues/34978 (1.13.4)
- [x] build/push cross-build image for 1.13.4
- [x] resolve BUILD file issue referencing crypto/ed25519 - https://github.com/bazelbuild/bazel-gazelle/issues/671
- [x] bump golang.org/x/crypto to go1.13 level
- [x] drop TMP commit allowing go1.12.x
- [x] update CI to run with go1.13.4 - https://github.com/kubernetes/test-infra/pull/15113
- [x] Fix flag parsing in node-e2e: https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/82809/pull-kubernetes-node-e2e/1191961293179850752/ and https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/82809/pull-kubernetes-node-e2e-containerd/1191961293142102016/
- [x] update CI to use go1.13.4 images - https://github.com/kubernetes/test-infra/pull/15131
- [x] Fix go version / image used by pull-kubernetes-integration - https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/82809/pull-kubernetes-integration/1191961293163073537/ - https://github.com/kubernetes/test-infra/pull/15144

**Which issue(s) this PR fixes**:
xref https://github.com/kubernetes/kubernetes/issues/82531

**Does this PR introduce a user-facing change?**:
```release-note
Kubernetes now requires go1.13.4+ to build
```

/cc @fejta @cblecker @BenTheElder @thockin

### CODE REVIEW

**Review by fejta — APPROVED** [review]
/approve
/lgtm
/hold

**Comment by BenTheElder** [comment]
```
2019-09-18 03:00:27.555724 I | integration: terminated 1795862953965652868 (unix://localhost:17958629539656528680)
--- FAIL: TestValidOpenAPISpec (1.42s)
panic: Invalid schema provided to SchemaValidator: object has no key "default" [recovered]
	panic: Invalid schema provided to SchemaValidator: object has no key "default"
```
https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/82809/pull-kubernetes-bazel-test/1174153992704364545
hmm
/retest

File: build/build-image/cross/VERSION [hunk 1]
```diff
@@ -1 +1 @@
-v1.12.9-1
+v1.13.0-1
```
Comment (BenTheElder) [inline build/build-image/cross/VERSION]: are you planning to push this image?

Comment (liggitt) [reply build/build-image/cross/VERSION]: was waiting for the upstream scale issue to be resolved, and 1.13.1 to be released before messing with pushing images

Comment (BenTheElder) [reply build/build-image/cross/VERSION]: ACK

**Comment by cblecker** [comment]
/lgtm cancel

This PR won't go out as is, as we need go1.13.1

File: build/build-image/cross/VERSION (hunk 1 above)
Comment (zouyee) [reply build/build-image/cross/VERSION]: 1.13.1 is out.

**Comment by BenTheElder** [comment]
https://github.com/golang/go/issues/32828#issuecomment-535560654 we still need to wait for 1.13.2

**Comment by liggitt** [comment]
go1.13.3 has been released. @wojtek-t is that on the scalability team's radar to verify the perf regression fix on?

**Comment by wojtek-t** [comment]
@mm4tt - we should verify the go1.13.3 to check if everything that we needed was cherrypicked there.

**Comment by mm4tt** [comment]
I can confirm that golang1.13.3 contains all scalability patches, more details can be found in https://github.com/golang/go/issues/34556 and https://github.com/golang/go/issues/34149.

I think we can go with merging this PR and verifying in our 5K CI/CD tests. We're currently impacted by https://github.com/kubernetes/kubernetes/issues/84151 and the tests are failing but a fix has been already merged and we have high hopes that today's run will pass.

Given that, can we wait till tomorrow with submitting this?

**Comment by zouyee** [comment]
At first, `golang version` may be updated in ci-image.
https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/82809/pull-kubernetes-typecheck/1187254464956862465

**Comment by liggitt** [comment]
@cblecker @BenTheElder sounds like we're ready to do this. what's the process for building/pushing the cross-build image?

**Comment by BenTheElder** [comment]
Someone needs to update the image sources for v1.13.3.
A googler needs to do make -C build/build-image/cross push and then file a
promotion CL internally for the resulting image.

**Comment by liggitt** [comment]
seeing a reproducible unit test panic in openapi swagger parsing. digging :(


the following passes on 1.12 and panics on 1.13:
```
import (
  "github.com/go-openapi/validate"
  "github.com/go-openapi/spec"
  "github.com/go-openapi/strfmt"
)

validate.NewSchemaValidator(spec.MustLoadSwagger20Schema(), nil, "", strfmt.Default)
```

**Comment by liggitt** [comment]
looks like the reflect.IsZero() added in go1.13 breaks go-openapi/pointer

**Comment by josiahbjorgaard** [comment]
Bug triage for 1.17 here. Code freeze is Nov. 18th. Will this PR be merged by then? If not, can we move it out of the milestone?

**Comment by BenTheElder** [comment]
Ideally this will merge by then.

**Comment by BenTheElder** [comment]
needs rebase

**Comment by liggitt** [comment]
rebased

**Comment by cblecker** [comment]
/test all
Anything that uses the cross build image or bazel should pass. Things like verify might fail. Let's get some better signal.

**Comment by liggitt** [comment]
There's an issue with the BUILD file generated for golang.org/x/crypto referring to a new stdlib package in go1.13 under vendor

**Comment by BenTheElder** [comment]
other than the temp commit at the end:
/lgtm
/approve

**Comment by liggitt** [comment]
two green runs of e2e/unit/integration tests at https://prow.k8s.io/pr-history/?org=kubernetes&repo=kubernetes&pr=82809 on e892267 and a3cea6c

dropped temp commit, ensured `hack/verify-typecheck.sh` and `hack/verify-vendor.sh` passed locally

**Comment by cblecker** [comment]
/hold
Holding to block retesting

**Comment by liggitt** [comment]
Added the links to the failed CI runs for those three jobs to the description. I'll look to sig-testing folks to help understand what needs to be adjusted for those jobs..

**Comment by BenTheElder** [comment]
/lgtm
/approve
:shipit:

**Comment by cblecker** [comment]
/lgtm
/approve
/hold cancel
SHIP IT

**Comment by liggitt** [comment]
Current batch merged and this is alone in the merge queue. Restarting the test to test against master HEAD

/test all

**Pull request merged**